File: public/docs/_examples/intro/ts/src/app/backend.service.ts
import {Injectable} from 'angular2/core';
import {Logger} from './logger.service';
import {Hero} from './hero';

@Injectable()
export class BackendService {
  constructor(private _logger: Logger) {}

  getAll<T>(type: {new(...args:any[]): any }) : any[]{
    if (type === Hero) {
      // TODO get from the database and return as a promise
      return [
        new Hero('Windstorm', 'Weather mastery'),
        new Hero('Mr. Nice', 'Killing them with kindness'),
        new Hero('Magneta', 'Manipulates metalic objects')];
    }
    let err = new Error('Cannot get object of this type');
    this._logger.error(err);
    throw err;
  }
}

File: public/docs/_examples/intro/dart/lib/backend_service.dart
// #docregion
library developer_guide_intro.backend_service;

import 'package:angular2/angular2.dart';
import 'package:developer_guide_intro/logger_service.dart';
import 'package:developer_guide_intro/hero.dart';

@Injectable()
class BackendService {
  final Logger _logger;
  List getAll(type) {
    // TODO get from the database and return as a promise
    if (type == Hero) {
      return [
        new Hero('Windstorm', power: 'Weather mastery'),
        new Hero('Mr. Nice', power: 'Killing them with kindness'),
        new Hero('Magneta', power: 'Manipulates metalic objects')
      ];
    }
    _logger.error('Cannot get object of this type');
    throw new ArgumentError("TODO: put log content here");
  }

  BackendService(Logger this._logger);
}
